devices/power.py

```diff
@@ -64,6 +64,8 @@ def get_power_device(ip_address, username=None, password=None, outlet=None):
         return Ip9258(ip_address, outlet, username=username, password=password)
     if 'Cyber Power Systems' in data:
         return CyberPowerPdu(ip_address, outlet=outlet, username=username, password=password)
+    if 'Unauthorized' in data:
+        return Ip9820(ip_address, outlet)
```

Comment:
Why Unauthorized? That might match other http data in other devices. Can you please share the full output?

Reply:
We can't get power device name to mask sure which power device in use.
Only can get below message, so we use 'Unauthorized' to check power device is Ip9820.

Please refer below http data message:

<html><head><title>Document Error: Unauthorized</title></head>
--
  | <body><h2>Access Error: Unauthorized</h2>
  | <p>Access to this document requires a User ID</p></body></html>

Reply:
I understand that, but I think you need to try to login and see what model this is... if you just do Unauthorized you will match ANY device that has that text in their output. Well that's more than just this device. You need to think generically.

I would say, if you don't match anything without logging in- then you do just that login and see if you can find a model number in the output and match off that.